File: algorithm/divisor/divisor.php
<?php

$N = 36;    // 整数 N

printf("1." . PHP_EOL);
$out = "{";
for($i = 1; $i <= $N; $i++) {
    if($N % $i == 0) {
        $out .= $i . ",";
    }
}
$out .= "}";
printf($out . PHP_EOL);

printf("2." . PHP_EOL);
$a = array();
for($i = 1; $i * $i <= $N; $i++) {
    if($N % $i == 0) {
        array_push($a, $i);
        if($i * $i != $N) {
            array_push($a, $N / $i);
        }
    }
}
sort($a);
$out = "{";
foreach ($a as $v) {
    $out .= $v . ",";
}
$out .= "}";
printf($out . PHP_EOL);

printf("2'." . PHP_EOL);
$asc = array();     // 昇順
$desc  = array();   // 降順
for($i = 1; $i * $i <= $N; $i++) {
    if($N % $i == 0) {
        array_push($asc, $i);
        if($i * $i != $N) {
            array_unshift($desc, $N / $i);
        }
    }
}
$out = "{";
foreach ($asc as $v) {
    $out .= $v . ",";
}
foreach ($desc as $v) {
    $out .= $v . ",";
}
$out .= "}";
printf($out . PHP_EOL);

?>

File: algorithm/divisor/divisor.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import math

N = 36      # 整数 N

print("1.")
out = "{"
for i in range(1, N+1):
    if N % i == 0:
        out += (str(i)) + ","
out += "}"
print(out)

print("2.")
arr = []
max = math.ceil(math.sqrt(N+1)) + 1
for i in range(1, max):
    if N % i == 0:
        arr.append(i)
        if i**2 != N:
            arr.append(N // i)
arr.sort()
out = "{"
for v in arr:
    out += (str(v)) + ","
out += "}"
print(out)

print("2'.")
asc = []        # 昇順
desc = []       # 降順
max = math.ceil(math.sqrt(N+1)) + 1
for i in range(1, max):
    if N % i == 0:
        asc.append(i)
        if i**2 != N:
            desc.insert(0, N // i)
out = "{"
for v in asc:
    out += (str(v)) + ","
for v in desc:
    out += (str(v)) + ","
out += "}"
print(out)
